flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphVariousFailuesTest.java

```diff
@@ -20,86 +20,31 @@
 
-
 public class ExecutionGraphVariousFailuesTest extends TestLogger {
 
-	/**
-	 * Test that failing in state restarting will retrigger the restarting logic. This means that
-	 * it only goes into the state FAILED after the restart strategy says the job is no longer
-	 * restartable.
-	 */
-	@Test
-	public void testFailureWhileRestarting() throws Exception {
-		final ExecutionGraph eg = ExecutionGraphTestUtils.createSimpleTestGraph(new InfiniteDelayRestartStrategy(2));
-		eg.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-		eg.scheduleForExecution();
-
-		assertEquals(JobStatus.RUNNING, eg.getState());
-		ExecutionGraphTestUtils.switchAllVerticesToRunning(eg);
-
-		eg.failGlobal(new Exception("Test 1"));
-		assertEquals(JobStatus.FAILING, eg.getState());
-		ExecutionGraphTestUtils.completeCancellingForAllVertices(eg);
-
-		// we should restart since we have two restart attempts left
-		assertEquals(JobStatus.RESTARTING, eg.getState());
-
-		eg.failGlobal(new Exception("Test 2"));
-
-		// we should restart since we have one restart attempts left
-		assertEquals(JobStatus.RESTARTING, eg.getState());
-
-		eg.failGlobal(new Exception("Test 3"));
-
-		// after depleting all our restart attempts we should go into Failed
-		assertEquals(JobStatus.FAILED, eg.getState());
-	}
-
-	/**
-	 * Tests that a {@link SuppressRestartsException} in state RESTARTING stops the restarting
-	 * immediately and sets the execution graph's state to FAILED.
-	 */
-	@Test
-	public void testSuppressRestartFailureWhileRestarting() throws Exception {
-		final ExecutionGraph eg = ExecutionGraphTestUtils.createSimpleTestGraph(new InfiniteDelayRestartStrategy(10));
-		eg.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-		eg.scheduleForExecution();
-
-		assertEquals(JobStatus.RUNNING, eg.getState());
-		ExecutionGraphTestUtils.switchAllVerticesToRunning(eg);
-
-		eg.failGlobal(new Exception("test"));
-		assertEquals(JobStatus.FAILING, eg.getState());
-
-		ExecutionGraphTestUtils.completeCancellingForAllVertices(eg);
-		assertEquals(JobStatus.RESTARTING, eg.getState());
-
-		// suppress a possible restart
-		eg.failGlobal(new SuppressRestartsException(new Exception("Test")));
-
-		assertEquals(JobStatus.FAILED, eg.getState());
-	}
-
 	/**
 	 * Tests that a failing scheduleOrUpdateConsumers call with a non-existing execution attempt
 	 * id, will not fail the execution graph.
 	 */
 	@Test
 	public void testFailingScheduleOrUpdateConsumers() throws Exception {
-		final ExecutionGraph eg = ExecutionGraphTestUtils.createSimpleTestGraph(new InfiniteDelayRestartStrategy(10));
-		eg.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-		eg.scheduleForExecution();
+		final SchedulerBase scheduler = SchedulerTestingUtils.newSchedulerBuilder(new JobGraph()).build();
+		scheduler.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
+		scheduler.startScheduling();
+
+		final ExecutionGraph eg = scheduler.getExecutionGraph();
```

Comment:
`scheduleOrUpdateConsumers` is still used by lazy-from-sources scheduling with schedulerNG. So we still need this test.
It is not needed by pipelined region scheduling though. Actually the partition cache mechanism is not needed anymore for pipelined region scheduling.
But maybe we still need keep this mechanism for other possible scheduling strategy which may schedule a task before all its upstream task finishes.
WDYT?

Reply:
Would it make sense to annotate tests which are specifically written for the non-pipelined region scheduler with a special annotation? That way it would be much easier to find them when we throw out the old scheduling strategies.

Reply:
@tillrohrmann Sure we can do that. Maybe `@TestLegacyScheduling`?
I will go through the modified tests to try to find out them.

Reply:
I checked the tests and find only 2 related to legacy eager/lazy scheduling. However, I think I can directly remove or rework them because they are testing other things but just was designed to use eager/lazy scheduling. In this way, I think we do not need an annotation like `@TestLegacyScheduling` anymore.
- ExecutionGraphNotEnoughResourceTest: can be removed because it is superseded by `DefaultSchedulerTest#failJobIfNotEnoughResources()`
- ExecutionGraphDeploymentTest#testEagerSchedulingWaitsOnAllInputPreferredLocations(): added a new test `DefaultSchedulerTest#deployTasksOnlyWhenAllSlotRequestsAreFulfilled()` to replace it

Regarding tests for `scheduleOrUpdateConsumers`, I think the `scheduleOrUpdateConsumers` mechanism deserves a separate discussion so I opened FLINK-20439 to track and discuss it.

Reply:
@azagrebin I see. Agreed it's better to test upon `SchedulerBase::scheduleOrUpdateConsumers`.

Reply:
`ExecutionGraphDeploymentTest#testEagerSchedulingWaitsOnAllInputPreferredLocations` seems to test that consumers wait on producers. The JobGraph has only one vertex in `DefaultSchedulerTest#deployTasksOnlyWhenAllSlotRequestsAreFulfilled` and it seems to test a different thing. Is it intended?

Reply:
Yes exactly as Till says.